scripts/NewznabToCardigannYML.ps1

```diff
@@ -35,208 +48,273 @@ function Invoke-NameReplace {
+    $catName = Invoke-CatNameReplace -Name $category.name
     $temp = [PSCustomObject][ordered]@{
-        id = $category.id
-        cat = $catName
-        desc = $category.description
+        id   = $category.id
+        cat  = $catName
+        desc = "'$($category.name)'"
     }
     $categories.Add($temp)
-}
-
-foreach ($category in $rCategories.subcat)
-{
-    $catName = Invoke-NameReplace -Name "$($category.description)/$($category.name)"
-    $temp = [PSCustomObject][ordered]@{
-        id = $category.id
-        cat = $catName
-        desc = $category.description
+    Write-Information 'Building Sub-Categories'
+    foreach ($subcategory in $category.subcat)
+    {
+        $subcatName = Invoke-CatNameReplace -Name "$($catName)/$($subcategory.name)"
+        $subtemp = [PSCustomObject][ordered]@{
+            id   = $subcategory.id
+            cat  = $subcatName
+            desc = "'$($catName)/$($subcategory.name)'"
+        }
+        $categories.Add($subtemp)
     }
-    $categories.Add($temp)
 }
-
+Write-Information 'Categories Built'
 [System.Collections.Generic.List[string]]$ymlCategories = @()
 foreach ($category in ($categories | Sort-Object id))
 {
     # TODO: This is currently causing the output to be a list of strings
     $ymlCategories.Add("{ id: $($category.id), cat: $($category.cat), desc: $($category.desc) }")
 }
-
-# ToDo: Validate Categories List (Names) - use newznabcats.txt
-# None matching categories (case insensitive) will need to be commented out
+# TODO: Validate Categories List (Names) - use newznabcats.txt
+# None matching categories (case insensitive) will need to be commented out - fuzzy match if possible?
+Write-Information 'Categories converted to YML'
 
 #TODO: This is currently creating strings for each mode and these shouldn't be strings
 $modes = [ordered]@{
     search = $q_search
 }
 
-if ($tv_search -ne "[]")
+if ($tv_search -ne '[]')
 {
     $modes['tv-search'] = $tv_search
 }
-if ($movie_search -ne "[]")
+if ($movie_search -ne '[]')
 {
     $modes['movie-search'] = $movie_search
 }
-if ($book_search -ne "[]")
+if ($book_search -ne '[]')
 {
     $modes['book-search'] = $book_search
 }
-if ($audio_search -ne "[]")
+if ($audio_search -ne '[]')
 {
     $modes['audio-search'] = $audio_search
 }
 
-#TODO: Finish this off with some if statements as per modes
 $inputs = [ordered]@{
-    t = "{{ .Query.Type }}"
-    apikey = "{{ .Config.apikey }}"
-    q = "{{ .Keywords }}"
+    t      = '{{ .Query.Type }}'
+    apikey = '{{ .Config.apikey }}'
+    q      = '{{ .Keywords }}'
+}
+
+foreach ($searchinput in ($modes.GETENUMERATOR() | ForEach-Object { $_.VALUE }).Replace('q', '').Replace('[', '').Replace(']', '').Split(',') | Sort-Object)
+{
+    if ($searchinput -ne 'q' -and $searchinput -ne '')
+    {
+        #return $searchinput
+        [string]$searchstring = $(Invoke-ModesReplace -Name $($searchinput))
+        $inputs.Add($($searchinput), "{{ .Query.$($searchstring)}}")
```

Comment:
You have said this is getting some elements twice - just do an if inputs doesn't contain on the key

Reply:
> C:\Development\Code\Prowlarr_Indexers\scripts\NewznabToCardigannYML.ps1 -site "https://www.tabula-rasa.pw" -indexer "Tabula Rasa" -privacy "private" -apipath "/api/v1/api"-outputfile "C:\Development\Code\Prowlarr_Indexers\definitions\v4\tabularasa.yml"